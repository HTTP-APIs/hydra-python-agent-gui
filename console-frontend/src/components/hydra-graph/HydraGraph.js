import React from "react";
import { withStyles } from "@material-ui/styles";
import { DataSet, Network } from "visjs-network";

const styles = (theme) => ({
  graphContainer: {
    height: "500px",
    backgroundColor: "#FFF",
    margin: "0 1em",
    borderRadius: "8px",
    width: "90%",
  },
  graphName: {
    color: "black",
    border: "1px solid white",
    borderRadius: "8px",
    width: "100%",
    letterSpacing: "2px",
  },
});

class HydraGraph extends React.Component {
  componentDidMount() {
    const self = this;
    // Create Node and Edge Datasets
    const nodes = new DataSet(this.props.apidocGraph.nodes);
    const edges = new DataSet(this.props.apidocGraph.edges);
    // Get reference to the mynetwork div
    const container = document.getElementById("mynetwork");

    const data = {
      nodes: nodes,
      edges: edges,
    };

    // See vis.js network options for more details on how to use this
    const options = {
      interaction: { hover: true },
      autoResize: true,
      nodes: {
        color: {
          hover: {
            border: "#5BDE79",
            background: "#5BDE79",
          },
        },
      },
      layout: {
        improvedLayout: true,
      },
    };
    // Create a network
    // eslint-disable-next-line
    let endpoint;
    let check = 0;
    let endpoints = null;

    for (const index in this.props.hydraClasses) {
      if (this.props.hydraClasses[index]["@id"] === "vocab:EntryPoint") {
        endpoints = this.props.hydraClasses[index].supportedProperty;
      }
    }

    const network = new Network(container, data, options);
    this.selectedNode = function (e) {
      this.props.selectNode(e);
    };
    network.on("hoverNode", function (event) {
      check = 0;
      const node = event.node;
      const element_array = Object.keys(data.nodes._data).map(function (key) {
        return data.nodes._data[key];
      });

      element_array.forEach((element) => {
        if (element.id === node) {
          endpoint = element;
          endpoints.forEach((ept) => {
            if (ept.property.label === endpoint.label) {
              check = 1;
            }
          });
        }
      });

      if (check !== 1) {
        const edges_array = Object.keys(data.edges._data).map(function (key) {
          return data.edges._data[key];
        });

        edges_array.forEach((edge) => {
          if (edge.to === node && edge.label === "supportedOp") {
            element_array.forEach((element) => {
              if (element.id === edge.from) {
                endpoint = element;
                endpoints.forEach((ept) => {
                  if (ept.property.label === endpoint.label) {
                    check = 1;
                  }
                });
              }
            });
          }
        });
      }

      if (check === 1) {
        options.nodes.color.hover.background = "#5BDE79";
        options.nodes.color.hover.border = "#5BDE79";
        network.setOptions(options);
      } else {
        options.nodes.color.hover.background = "#FBD20B";
        options.nodes.color.hover.border = "#FBD20B";
        network.setOptions(options);
      }
    });

    network.on("select", function (event) {
      check = 0;
      let selectedRequest;
      const { nodes } = event;
      const element_array = Object.keys(data.nodes._data).map(function (key) {
        return data.nodes._data[key];
      });

      element_array.forEach((element) => {
        if (element.id === nodes[0]) {
          endpoint = element;
        }
      });

      let i = 0;
      endpoints.forEach((endpoints) => {
        if (endpoints.property.label === endpoint.label) {
          check = 1;
          selectedRequest = { Index: i, operation: "GET" };
          self.selectedNode(selectedRequest);
        }
        i += 1;
      });
      if (check !== 1) {
        const operation = endpoint.label;
        const edges_array = Object.keys(data.edges._data).map(function (key) {
          return data.edges._data[key];
        });

        edges_array.forEach((edge) => {
          if (edge.to === nodes[0] && edge.label === "supportedOp") {
            element_array.forEach((element) => {
              if (element.id === edge.from) {
                endpoint = element;
                i = 0;
                endpoints.forEach((endpoints) => {
                  if (endpoints.property.label === endpoint.label) {
                    check = 1;
                    selectedRequest = { Index: i, operation: operation };
                    self.selectedNode(selectedRequest);
                  }
                  i += 1;
                });
              }
            });
          }
        });
      }
    });
    network.moveTo({
      scale: 1.3,
      animation: {
        duration: "0.5s",
        easingFunction: "linear",
      },
    });
  }

  render() {
    const { classes } = this.props;
    return (
      <header className="app-header">
        <div className={classes.graphContainer} id="mynetwork"></div>
      </header>
    );
  }
}

export default withStyles(styles)(HydraGraph);
