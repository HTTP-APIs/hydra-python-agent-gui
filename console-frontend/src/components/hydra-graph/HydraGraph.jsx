import React from 'react'
import { withStyles } from '@material-ui/styles';
// eslint-disable-next-line
import { DataSet, Network } from 'visjs-network';

const styles = theme => ({
    graphContainer: {
        width: '100%',
        height: '82vh',
        backgroundColor: '#EEEEEE',
    },
});

class HydraGraph extends React.Component {

    componentDidMount(){
        debugger
        var { DataSet, Network } = require('visjs-network');

        // Create Node and Edge Datasets
        var nodes = new DataSet(this.props.apidocGraph.nodes)
        var edges = new DataSet(this.props.apidocGraph.edges)

        // Get reference to the mynetwork div
        var container = document.getElementById('mynetwork');

        var data = {
            nodes: nodes,
            edges: edges
        };

        // See vis.js network options for more details on how to use this
        var options = {};
        // Create a network
        // eslint-disable-next-line
        var network = new Network(container, data, options);
    }

   render() {
    const { classes } = this.props;
       return (
            <header className="app-header">
                <div className={classes.graphContainer} id="mynetwork"></div>
            </header>
       )
    }
}

export default withStyles(styles)(HydraGraph);
