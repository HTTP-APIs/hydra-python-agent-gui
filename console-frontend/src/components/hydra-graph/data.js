export const nodes = [{"id":1,"label":"Entrypoint","shape":"hexagon","size":15},{"color":{"background":"#FBD20B"},"id":2,"label":"Area","shape":"box","size":12},{"color":{"background":"#FBD20B"},"id":3,"label":"POST","shape":"circle","size":10},{"color":{"background":"#FBD20B"},"id":4,"label":"Area","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":5,"label":"TopLeft","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":6,"label":"BottomRight","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":7,"label":"null","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":8,"label":"GET","shape":"circle","size":10},{"color":{"background":"#FBD20B"},"id":9,"label":"null","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":10,"label":"Area","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":11,"label":"TopLeft","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":12,"label":"BottomRight","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":13,"label":"StateCollection","shape":"box","size":12},{"color":{"background":"#FBD20B"},"id":14,"label":"GET","shape":"circle","size":10},{"color":{"background":"#FBD20B"},"id":15,"label":"null","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":16,"label":"StateCollection","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":17,"label":"PUT","shape":"circle","size":10},{"color":{"background":"#FBD20B"},"id":18,"label":"State","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":19,"label":"Speed","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":20,"label":"Position","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":21,"label":"Direction","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":22,"label":"Battery","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":23,"label":"SensorStatus","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":24,"label":"DroneID","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":25,"label":"State","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":26,"label":"Speed","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":27,"label":"Position","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":28,"label":"Direction","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":29,"label":"Battery","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":30,"label":"SensorStatus","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":31,"label":"DroneID","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":32,"label":"CommandCollection","shape":"box","size":12},{"color":{"background":"#FBD20B"},"id":33,"label":"GET","shape":"circle","size":10},{"color":{"background":"#FBD20B"},"id":34,"label":"null","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":35,"label":"CommandCollection","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":36,"label":"PUT","shape":"circle","size":10},{"color":{"background":"#FBD20B"},"id":37,"label":"Command","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":38,"label":"DroneID","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":39,"label":"State","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":40,"label":"Command","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":41,"label":"DroneID","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":42,"label":"State","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":43,"label":"MessageCollection","shape":"box","size":12},{"color":{"background":"#FBD20B"},"id":44,"label":"GET","shape":"circle","size":10},{"color":{"background":"#FBD20B"},"id":45,"label":"null","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":46,"label":"MessageCollection","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":47,"label":"PUT","shape":"circle","size":10},{"color":{"background":"#FBD20B"},"id":48,"label":"Message","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":49,"label":"MessageString","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":50,"label":"Message","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":51,"label":"MessageString","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":52,"label":"DatastreamCollection","shape":"box","size":12},{"color":{"background":"#FBD20B"},"id":53,"label":"GET","shape":"circle","size":10},{"color":{"background":"#FBD20B"},"id":54,"label":"null","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":55,"label":"DatastreamCollection","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":56,"label":"PUT","shape":"circle","size":10},{"color":{"background":"#FBD20B"},"id":57,"label":"Datastream","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":58,"label":"Temperature","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":59,"label":"DroneID","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":60,"label":"Position","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":61,"label":"Datastream","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":62,"label":"Temperature","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":63,"label":"DroneID","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":64,"label":"Position","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":65,"label":"DroneCollection","shape":"box","size":12},{"color":{"background":"#FBD20B"},"id":66,"label":"GET","shape":"circle","size":10},{"color":{"background":"#FBD20B"},"id":67,"label":"null","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":68,"label":"DroneCollection","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":69,"label":"PUT","shape":"circle","size":10},{"color":{"background":"#FBD20B"},"id":70,"label":"Drone","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":71,"label":"DroneState","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":72,"label":"name","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":73,"label":"model","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":74,"label":"MaxSpeed","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":75,"label":"Sensor","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":76,"label":"Drone","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":77,"label":"DroneState","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":78,"label":"name","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":79,"label":"model","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":80,"label":"MaxSpeed","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":81,"label":"Sensor","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":82,"label":"LogEntryCollection","shape":"box","size":12},{"color":{"background":"#FBD20B"},"id":83,"label":"GET","shape":"circle","size":10},{"color":{"background":"#FBD20B"},"id":84,"label":"null","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":85,"label":"LogEntryCollection","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":86,"label":"PUT","shape":"circle","size":10},{"color":{"background":"#FBD20B"},"id":87,"label":"LogEntry","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":88,"label":"DroneID","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":89,"label":"Update","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":90,"label":"Get","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":91,"label":"Send","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":92,"label":"State","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":93,"label":"Data","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":94,"label":"Command","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":95,"label":"LogEntry","shape":"circle","size":8},{"color":{"background":"#FBD20B"},"id":96,"label":"DroneID","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":97,"label":"Update","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":98,"label":"Get","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":99,"label":"Send","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":100,"label":"State","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":101,"label":"Data","shape":"box","size":7},{"color":{"background":"#FBD20B"},"id":102,"label":"Command","shape":"box","size":7}]
export const edges = [{"from":1,"to":2,"id":"f0c421cf-9f25-43e2-842d-91ca2480b0bc"},{"from":2,"label":"supportedOp","to":3,"id":"8b4af3e7-b282-4907-94d1-98bfd101b681"},{"from":3,"label":"expects","to":4,"id":"636d2aa1-415e-41c3-84a5-be28ab015928"},{"from":4,"label":"supportedProp","to":5,"id":"18633578-1c17-459c-8c6f-ad344a931124"},{"from":4,"label":"supportedProp","to":6,"id":"22baab83-a5a4-431c-8c86-a88a20e44081"},{"from":3,"label":"returns","to":7,"id":"1be75de6-6de4-46f7-ad81-db06aa1aebf9"},{"from":2,"label":"supportedOp","to":8,"id":"ec86000c-c6b1-4b95-a37b-092c383e5175"},{"from":8,"label":"expects","to":9,"id":"c3b245ff-32c6-4245-8c23-822ddd5d8945"},{"from":8,"label":"returns","to":10,"id":"991ea58d-5c51-4dbb-8e2b-0abbf18de1af"},{"from":10,"label":"supportedProp","to":11,"id":"ef900203-7b93-48c9-b562-99ef8e801f21"},{"from":10,"label":"supportedProp","to":12,"id":"b3414e8e-18f8-4ff5-aff3-9d4eeeefee85"},{"from":1,"to":13,"id":"ee7c7584-8d5c-461a-89a0-8090fcc06aaf"},{"from":13,"label":"supportedOp","to":14,"id":"fcc57e06-b35e-413d-a204-bfb16a073d87"},{"from":14,"label":"expects","to":15,"id":"8f96ce62-d4fd-439b-94a1-300fcbda3781"},{"from":14,"label":"returns","to":16,"id":"e75dde6a-277a-4c98-9fb3-88af8bf37596"},{"from":13,"label":"supportedOp","to":17,"id":"0340d0b2-2bf0-496c-b0c9-6cdcba4a9bb9"},{"from":17,"label":"expects","to":18,"id":"4581df75-b851-4c1d-833e-b2c5e20ebbec"},{"from":18,"label":"supportedProp","to":19,"id":"1b2c7ba3-f7fa-4c75-a40f-0c274a74c1a1"},{"from":18,"label":"supportedProp","to":20,"id":"f7f065f3-84aa-444b-a662-f3ac53f664ed"},{"from":18,"label":"supportedProp","to":21,"id":"40221f20-4041-41e2-a934-432f72cd9d82"},{"from":18,"label":"supportedProp","to":22,"id":"7d449fe7-f04f-4586-ac52-2de97523ab1a"},{"from":18,"label":"supportedProp","to":23,"id":"b628a033-8b1f-419a-9f1f-adf00519e17b"},{"from":18,"label":"supportedProp","to":24,"id":"f2663264-8612-40a2-8e85-1ea92b869094"},{"from":17,"label":"returns","to":25,"id":"556ae3ee-6e9f-4b5c-b14d-0cb6fccefa18"},{"from":25,"label":"supportedProp","to":26,"id":"07a0ac3c-5070-4199-a8ba-3a59adf6ad15"},{"from":25,"label":"supportedProp","to":27,"id":"54185b17-f1db-42d0-a95d-1c27200a817a"},{"from":25,"label":"supportedProp","to":28,"id":"2ef0f225-5576-4e68-bf99-efeabe8947de"},{"from":25,"label":"supportedProp","to":29,"id":"1ece9f3e-74bd-44f0-a936-d4a2d3f047b2"},{"from":25,"label":"supportedProp","to":30,"id":"a6660873-2872-4cfe-b5da-d234cf37d1be"},{"from":25,"label":"supportedProp","to":31,"id":"43de3796-d82f-4703-b4bf-6e1dc0c41ecc"},{"from":1,"to":32,"id":"7d186c23-fbdd-4601-b519-f70c4231a5f0"},{"from":32,"label":"supportedOp","to":33,"id":"f5b75a4c-7bec-4bc0-b172-8fdbeecb159f"},{"from":33,"label":"expects","to":34,"id":"ef7dc768-8f35-4efc-be50-699696dc1441"},{"from":33,"label":"returns","to":35,"id":"500d79fd-6984-4c50-9833-e89ac53e3fb8"},{"from":32,"label":"supportedOp","to":36,"id":"b7bd2d24-9e7b-43a1-af68-14b172e2ba73"},{"from":36,"label":"expects","to":37,"id":"49de838e-fb3a-47ef-a090-50653bc80c94"},{"from":37,"label":"supportedProp","to":38,"id":"99da3616-5ca9-4c38-86ad-37e8eee28108"},{"from":37,"label":"supportedProp","to":39,"id":"661a3026-90a4-4192-9f8b-44990297af4f"},{"from":36,"label":"returns","to":40,"id":"d77d0807-91e0-4834-aa86-4e1133672f2f"},{"from":40,"label":"supportedProp","to":41,"id":"683c402e-e3cc-4c66-932e-6a47ef57cc86"},{"from":40,"label":"supportedProp","to":42,"id":"188a47c5-dde4-4fff-a3ff-27c6c104a128"},{"from":1,"to":43,"id":"167f192d-a5bc-4eff-8fe0-4200dd97a698"},{"from":43,"label":"supportedOp","to":44,"id":"8f16f408-1ff3-415a-ba73-1ce3db6be1f8"},{"from":44,"label":"expects","to":45,"id":"42c113c8-e3dc-464c-b308-49c87d92eeb5"},{"from":44,"label":"returns","to":46,"id":"bf214dbb-46df-4e3e-87a7-d2695d5dd0db"},{"from":43,"label":"supportedOp","to":47,"id":"057587fb-4bb4-4687-86ec-bdbc32e8d883"},{"from":47,"label":"expects","to":48,"id":"5b1246d8-47f7-49ce-86e8-caf03420bd16"},{"from":48,"label":"supportedProp","to":49,"id":"cda8f99f-41d2-4751-811f-f1839f14cc1e"},{"from":47,"label":"returns","to":50,"id":"1305b06a-856e-4749-bce3-8147b2391519"},{"from":50,"label":"supportedProp","to":51,"id":"b1da1ab7-f63f-4188-bb69-b18c91ad9ad8"},{"from":1,"to":52,"id":"7eb406b4-43cf-4e79-87a6-db7cf1127c05"},{"from":52,"label":"supportedOp","to":53,"id":"0a779e65-4665-42d0-980f-d09f870c8245"},{"from":53,"label":"expects","to":54,"id":"4300b2f1-d651-414e-8cbf-8ddb4d875d2e"},{"from":53,"label":"returns","to":55,"id":"aac7bbfb-6c26-46ce-8fec-585f079d9584"},{"from":52,"label":"supportedOp","to":56,"id":"def57287-b53f-48c8-9a78-32d60af29924"},{"from":56,"label":"expects","to":57,"id":"a7c5d4b1-7dee-4be3-aa6b-6c053888d83d"},{"from":57,"label":"supportedProp","to":58,"id":"c5db8b08-ed3c-4b7d-b09f-a5232a93b2fe"},{"from":57,"label":"supportedProp","to":59,"id":"348187cd-6ac3-49d1-b142-5fe2b194f133"},{"from":57,"label":"supportedProp","to":60,"id":"c7442500-83e9-4f55-8a26-9450055eb395"},{"from":56,"label":"returns","to":61,"id":"0d280be3-cbcc-4bda-8885-7a873b63963d"},{"from":61,"label":"supportedProp","to":62,"id":"75d45a87-aa06-4c36-ad6a-6d8f30b608ef"},{"from":61,"label":"supportedProp","to":63,"id":"dea962ce-ff24-497a-84d2-6d0c8f133947"},{"from":61,"label":"supportedProp","to":64,"id":"f001aa9b-9c5d-4fd0-ac90-21a272bd60da"},{"from":1,"to":65,"id":"934efad1-d9c7-4ae7-8750-9557c537d80d"},{"from":65,"label":"supportedOp","to":66,"id":"ed233f1d-1547-48e9-9e03-6c4f62093749"},{"from":66,"label":"expects","to":67,"id":"0351dc58-3bb4-4260-a94c-c8e817049d45"},{"from":66,"label":"returns","to":68,"id":"9714e320-df2f-4342-99d8-60da361f841c"},{"from":65,"label":"supportedOp","to":69,"id":"c735d0bd-e6de-4ba8-a180-e83023abf25b"},{"from":69,"label":"expects","to":70,"id":"d7a54807-4ff3-4021-9616-c40b743ab18d"},{"from":70,"label":"supportedProp","to":71,"id":"3c5cd62a-19ff-4987-bba8-61ec8bb80e58"},{"from":70,"label":"supportedProp","to":72,"id":"dd6d05b1-e3f9-4a1c-a6ed-edc0d5c8bc8c"},{"from":70,"label":"supportedProp","to":73,"id":"a3fbe2a8-39e8-49a4-b782-fadeceddc674"},{"from":70,"label":"supportedProp","to":74,"id":"f776763e-2d25-4fe8-bc96-07f8e9dea49e"},{"from":70,"label":"supportedProp","to":75,"id":"b753e592-5f70-491a-8171-07b329905a78"},{"from":69,"label":"returns","to":76,"id":"470f866f-c211-44c3-9b94-c8a57c17a3e6"},{"from":76,"label":"supportedProp","to":77,"id":"3e818023-2de1-4949-919e-d1da7dd0f325"},{"from":76,"label":"supportedProp","to":78,"id":"a1396386-89c3-450b-8517-ef29f3252a97"},{"from":76,"label":"supportedProp","to":79,"id":"8e8df8f0-8f05-4ab1-90dc-fee8070defcc"},{"from":76,"label":"supportedProp","to":80,"id":"4f3805a3-7865-42e5-b209-73c08c2e31a3"},{"from":76,"label":"supportedProp","to":81,"id":"4c7a9c14-b4fe-43b1-8838-e8683db6e17f"},{"from":1,"to":82,"id":"dc576e9c-5dda-4e1f-8f33-4b740647e23d"},{"from":82,"label":"supportedOp","to":83,"id":"e5d416f4-a8c0-4811-92f5-b232f3af29af"},{"from":83,"label":"expects","to":84,"id":"23911ed5-a04d-4d63-93ad-dba47c0ebb24"},{"from":83,"label":"returns","to":85,"id":"27eabe2d-798b-4863-8063-8fa3d2488ad5"},{"from":82,"label":"supportedOp","to":86,"id":"aa6e6fc2-a616-49f7-af56-c3e6c55e9d97"},{"from":86,"label":"expects","to":87,"id":"114a41d0-1f55-4671-b52f-ff800f538ed1"},{"from":87,"label":"supportedProp","to":88,"id":"ed6656b4-c472-4849-8312-dc91cd277f74"},{"from":87,"label":"supportedProp","to":89,"id":"553db341-3699-4bb8-8093-50a8b47012be"},{"from":87,"label":"supportedProp","to":90,"id":"a6090a37-d022-4695-a984-6e3dc388a423"},{"from":87,"label":"supportedProp","to":91,"id":"2d161117-65f2-4fde-99d8-a7d4689d2cd9"},{"from":87,"label":"supportedProp","to":92,"id":"1100d181-e50a-41a3-bf9d-940460811a55"},{"from":87,"label":"supportedProp","to":93,"id":"78861769-8bda-4580-b27d-b6737cdcfb6d"},{"from":87,"label":"supportedProp","to":94,"id":"bf028695-2c8b-4d96-a368-cf81e4681ff4"},{"from":86,"label":"returns","to":95,"id":"26663c60-8850-445f-8f40-836542539dd1"},{"from":95,"label":"supportedProp","to":96,"id":"cf1f02ae-24e7-4200-bcca-c37541386cf3"},{"from":95,"label":"supportedProp","to":97,"id":"628c1f1b-94dd-4b60-9a5a-7c1a260c3bab"},{"from":95,"label":"supportedProp","to":98,"id":"669f7b20-e6bb-4e03-a007-c976cea0ca6f"},{"from":95,"label":"supportedProp","to":99,"id":"3a1c9c55-368b-446a-95b7-16d0979227bb"},{"from":95,"label":"supportedProp","to":100,"id":"95727a2c-0a42-4f82-9545-552e029490ee"},{"from":95,"label":"supportedProp","to":101,"id":"22601929-6b1f-4fc2-ac64-139d4e78f353"},{"from":95,"label":"supportedProp","to":102,"id":"43cb71b7-f8e9-4474-b146-a67f6fd424bd"}]