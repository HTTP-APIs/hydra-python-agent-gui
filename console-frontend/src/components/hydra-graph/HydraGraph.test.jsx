import React from 'react';
import HydraGraph from './HydraGraph';
import {nodes, edges} from './data';
import renderer from 'react-test-renderer';

describe('<HydraGraph />', () => {
  it('snapshot render', function() {
    const props = {
      apidocGraph: {
        nodes, edges
      }
    }
    const component = renderer.create(<HydraGraph {...props}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})
