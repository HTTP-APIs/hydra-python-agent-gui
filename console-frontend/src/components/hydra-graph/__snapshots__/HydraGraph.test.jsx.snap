// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HydraGraph /> snapshot render 1`] = `
<header
  className="app-header"
>
  <div
    className="HydraGraph-graphContainer-1"
    id="mynetwork"
  />
</header>
`;
